### PR TITLE
Fix -Wconversion warnings

### PR DESCRIPTION
**Pull request by 0x8000-0000**
I agree that my contributions are licensed under the {fmt} license, and agree to future changes to the licensing.

### CODE REVIEW

**Comment by 0x8000-0000** [comment]
This would close #1422

File: include/fmt/format.h [hunk 1]
```diff
@@ -1572,7 +1572,7 @@ template <typename Range> class basic_writer {
         it = format_decimal<char_type>(
             it, abs_value, size,
             [this, s, &group, &digit_index](char_type*& buffer) {
-              if (*group <= 0 || ++digit_index % *group != 0 ||
+              if (*group <= 0 || ++digit_index % static_cast<unsigned int>(*group) != 0 ||
```
Comment (vitaut) [inline include/fmt/format.h]: Please change `digit_index`'s type to `int` instead.

File: include/fmt/posix.h [hunk 2]
```diff
@@ -157,7 +157,7 @@ class buffered_file {
     other.file_ = nullptr;
   }
 
-  buffered_file& operator=(buffered_file&& other) FMT_NOEXCEPT {
+  buffered_file& operator=(buffered_file&& other) /* FMT_NOEXCEPT */ {
```
Comment (vitaut) [inline include/fmt/posix.h]: I suggest removing `/* FMT_NOEXCEPT */` altogether.

File: include/fmt/format.h (hunk 1 above)
Comment (0x8000-0000) [reply include/fmt/format.h]: OK

File: include/fmt/posix.h (hunk 2 above)
Comment (0x8000-0000) [reply include/fmt/posix.h]: OK

**Comment by vitaut** [comment]
Thanks!

**Pull request merged**